odxtools/cli/snoop.py

```diff
@@ -61,29 +62,30 @@ def handle_telegram(telegram_id, payload):
     else:
         print(f"Tester: "
               f"{payload.hex()} "
-              f"({payload}, {len(payload)} bytes)")
+              f"({payload!r}, {len(payload)} bytes)")
 
 
-def init_verbose_state_machine(BaseClass, *args, **kwargs):
+def init_verbose_state_machine(BaseClass: Type[IsoTpStateMachine], *args: Any,
+                               **kwargs: Any) -> IsoTpStateMachine:
 
-    class InformativeIsoTpDecoder(BaseClass):
+    class InformativeIsoTpDecoder(BaseClass):  # type: ignore[valid-type, misc]
```

Comment:
why?

Reply:
mypy has trouble understanding the class ought to be derived from the class type which is passed as a function argument. unfortunately, I couldn't find a good solution...